fs/fshttp/dialer.go

```diff
@@ -52,7 +52,16 @@ var warnDSCPFail, warnDSCPWindows sync.Once
 
 // DialContext connects to the network address using the provided context.
 func (d *Dialer) DialContext(ctx context.Context, network, address string) (net.Conn, error) {
-	c, err := d.Dialer.DialContext(ctx, network, address)
+	var c net.Conn
+	var err error
+	ci := fs.GetConfig(ctx)
+	if ci.PreferIPv4 {
```

Comment:
What are the possible values of `network` here? Maybe these if statements should be wrapped in an `if network == "tcp"`?

Reply:
The http transport will only use `tcp` which lets the dialer decide between IPv4 and IPv6. Other possible values would be `udp, udp4, udp6, ip, ip4, ip6`.